core/src/main/java/uk/gov/justice/services/core/interceptor/AsynchronousDispatchInterceptor.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.justice.services.core.interceptor;
+
+import uk.gov.justice.services.messaging.JsonEnvelope;
+
+import java.util.function.Consumer;
+
+/**
+ * Interceptor dispatch target for asynchronous calls to dispatch.
+ */
+public class AsynchronousDispatchInterceptor implements Interceptor {
+
+    private Consumer<JsonEnvelope> dispatcher;
+
+    public AsynchronousDispatchInterceptor(final Consumer<JsonEnvelope> dispatcher) {
+        this.dispatcher = dispatcher;
+    }
+
+    /**
+     * Calls the dispatcher with the input JsonEnvelope from the interceptor context.
+     *
+     * @param interceptorContext the context that contains the input JsonEnvelope
+     * @param interceptorChain   the interceptor chain, not used as this is the target of the chain
+     * @return the interceptor context passed to this method as there is no output from the
+     * dispatcher
+     */
+    @Override
+    public InterceptorContext process(final InterceptorContext interceptorContext, final InterceptorChain interceptorChain) {
```

Comment:
Having a not used parameter for a method implemented through an 'interface' contract doesn't have a good smell about it...